### PR TITLE
Again crystal invasion fix, why edition (+ balancing because why not)

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Hating on get_dist() proc returning -1 if the two turfs are on the same spot
Using now BETTER_DIST() a define i added to replace the native proc, it takes 3 arguments the distance var, the first atom and the second atom to check. if one of those atoms is missing, it returns -1 as error.

Little bit of balancing on the overall mob damage, and boss spawning

## Changelog
:cl:
add: BETTER_DIST define to replace the get_dist() native proc
fix: event crash fix by replacing the native proc with the define
tweak: monster damage overall reduced and boss spawning changed a bit
/:cl:

### CODE REVIEW

File: code/modules/events/crystal_event.dm [hunk 1]
```diff
@@ -832,7 +830,8 @@ This section is for the crystal monsters variations
 		var/mob/living/mob = clong
 		if(mob.stat >= HARD_CRIT)
 			mob.dust()
-			health += 35
+			if(health < 120)
+				health += 30
```
Comment (ATH1909) [inline code/modules/events/crystal_event.dm:834]:
```suggestion
			if(health < maHealth)
				health = min(health+30,maxHealth)
```

File: code/modules/events/crystal_event.dm [hunk 2]
```diff
@@ -844,7 +843,8 @@ This section is for the crystal monsters variations
 		var/mob/living/mob = target
 		if(mob.stat >= HARD_CRIT)
 			mob.dust()
-			health += 35
+			if(health < 120)
+				health += 30
```
Comment (ATH1909) [inline code/modules/events/crystal_event.dm:847]:
```suggestion
			if(health < maxHealth)
				health = min(health+30,maxHealth)
```

**Comment by Tlaltecuhtli** [comment]
are there any scaling based on population? i saw it happen on low pop (30ppl) and the few people who knew that you had to analyze the portal to close it got overwhelmed with like 10 new portals every 5 min

**Comment by Ghilker** [comment]
@Tlaltecuhtli there is no scaling based on the population, but maybe you saw an older version of it where there were more mobs spawning all around, the monster had high healt and delivered tons of damage. now the spawn has been nerfed as well as the damage overall.

**Pull request merged**